### PR TITLE
Fix selenium tests in projectexplorer package

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatoliy Bazko <abazko@redhat.com>

### What does this PR do?
Fix tests in projectexplorer package

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/10451

### CODE REVIEW

**Review by svor — APPROVED** [review]

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/DownloadProjectTest.java [hunk 1]
```diff
@@ -103,19 +104,27 @@ public void downloadAllProjectsTest() throws IOException {
...
-            "TestProject2/README.md",
+            "TestProject2/.classpath",
+            "TestProject2/.project",
+            "TestProject2/.settings/org.eclipse.jdt.apt.core.prefs",
+            "TestProject2/.settings/org.eclipse.jdt.core.prefs",
+            "TestProject2/.settings/org.eclipse.m2e.core.prefs",
             "TestProject2/pom.xml",
-            "TestProject2/src/main/java/org/eclipse/qa/examples/AppController.java",
-            "TestProject2/src/main/webapp/WEB-INF/jsp/guess_num.jsp",
+            "TestProject2/src/main/java.org.eclipse.qa.examples/AppController.java",
```
Comment (tsmaeder) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/DownloadProjectTest.java]: This does not look like a file from "/projects/ProjectWithDifferentTypeOfFiles2". Also, shouldn't we have a folder structure here?

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/DownloadProjectTest.java [hunk 2]
```diff
@@ -223,13 +233,17 @@ private String getDownloadedFileContent(String testFileName) throws IOException
...
+      throws IOException {
     IOException lastException = null;
     for (int i = 0; i < MAX_ATTEMPTS; i++) {
       try {
         return downloadedFileUtil
             .getPackageFileList(seleniumWebDriver, downloadedTestProject1PackageName)
-            .toString();
+            .stream()
+            // target directory appears asynchronously and leads to failed tests
+            .filter(s -> !s.contains("target"))
```
Comment (tsmaeder) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/DownloadProjectTest.java]: shouldn't this be "startsWith("/target") for more robustness?

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/dependencies/OpenExternalLibraryFileAfterRefreshTest.java [hunk 3]
```diff
@@ -86,6 +77,6 @@ private void openFileFromExternalLibraryAndCheckDisplayedText() {
 
     editor.waitActive();
     editor.waitTabIsPresent(CHECKING_FILE_NAME);
-    editor.waitTextIntoEditor(EXPECTED_EDITOR_TEXT);
+    assertTrue(editor.getVisibleTextFromEditor().contains("package javax.servlet"));
```
Comment (tsmaeder) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/dependencies/OpenExternalLibraryFileAfterRefreshTest.java:80]: This tests for less text than before. Why not check the whole text?

File: selenium/che-selenium-test/src/test/resources/projects/ProjectWithDifferentTypeOfFiles2/src/main/java.org.eclipse.qa.examples/AppController.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
+/*
```
Comment (tsmaeder) [inline selenium/che-selenium-test/src/test/resources/projects/ProjectWithDifferentTypeOfFiles2/src/main/java.org.eclipse.qa.examples/AppController.java]: Ok, I see now where the test above is coming from, But is this folder name even allowed in Java? If it isn't, shouldn't we reject this somehow?

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/editor/EditorAgentImpl.java [hunk 5]
```diff
@@ -650,6 +650,11 @@ public void apply(java.util.Optional<VirtualFile> optionalFile)
                           callback.onSuccess(null);
                         }
                       }
+                    })
+                .catchError(
+                    error -> {
```
[the remark this answers is not in the document]
Comment (tolusha) [reply ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/editor/EditorAgentImpl.java:655]: I don't think so.
Accordingly to the code above 
```
 } else {
                          removeFromOpeningFilesList(path, editorPartStack);
                          callback.onSuccess(null);
                        }
```
it is an usual case.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/dependencies/OpenExternalLibraryFileAfterRefreshTest.java (hunk 3 above)
Comment (tolusha) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/dependencies/OpenExternalLibraryFileAfterRefreshTest.java:80]: `waitTextIntoEditor` checks only visible text. So, it depends on the screen resolution.
In master branch it contains decomplied code which fits the screen. In 5730 branch it contains source code, that's why I've changed the expectation.

**Review by tsmaeder — APPROVED** [review]

**Pull request merged**